tests/e2e/utils/DriverHelper.ts

```diff
@@ -194,25 +201,34 @@ export class DriverHelper {
     }
 
     public async waitAllDisappearance(locators: Array<By>,
-        attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS,
-        polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
+        attemptsPerLocator: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS,
+        pollingPerLocator: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
 
         Logger.trace(`DriverHelper.waitAllDisappearance ${locators}`);
 
         for (const elementLocator of locators) {
-            await this.waitDisappearance(elementLocator, attempts, polling);
+            await this.waitDisappearance(elementLocator, attemptsPerLocator, pollingPerLocator);
         }
     }
 
     public async waitAndClick(elementLocator: By, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
-        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
         const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
 
         Logger.trace(`DriverHelper.waitAndClick ${elementLocator}`);
 
         for (let i = 0; i < attempts; i++) {
-            const element: WebElement = await this.waitVisibility(elementLocator, timeout);
-
+            let element: WebElement;
+            try {
+                element = await this.waitVisibility(elementLocator, polling);
+            } catch (err) {
+                if (i >= attempts) {
+                    Logger.error(`DriverHelper.waitAndClick - failed with exception, out of attempts - ${err}`);
+                    throw err;
+                } else {
+                    continue;
+                }
+            }
```

Comment:
Looks good, only one small improvement could you please add here and everywhere an empty row between try\catch blocks. In my opinion, it will increase readability. But, of course, if you have another opinion, feels free to ignore this message.